popde/density_estimate.py

```diff
@@ -93,6 +119,63 @@ def evaluate_scipy(self, points):
         return density_values
 
 
+    def plot_2d_contour(self, dim1, dim2, slice_dims=None, slice_values=None, file_name=None,num_points=100, **kwargs):
+        """
+        Plot a 2D contour of the KDE with optional slicing along other dimensions.
+
+        Parameters:
+            - dim1, dim2: Dimensions for the plot axes.
+            - slice_dims: Dimensions to slice along (list or tuple).
+            - slice_values: Values for slicing along slice_dims (list or tuple).
+            - num_points: Number of points for the contour plot.
+            - **kwargs: Additional keyword arguments passed to the `contour` function.
+        Example:
+            np.random.seed(42)
+            # Number of data points
+            num_points = 1000
+            # Mean and covariance matrix
+            mean = [0, 0, 0]
+            covariance_matrix = [[1, 0.5, 0.3],
+                                [0.5, 1, 0.2],
+                                [0.3, 0.2, 1]]
+
+            # Generate 3D normal distributed data
+            data = np.random.multivariate_normal(mean, covariance_matrix, num_points)
+            parameter = ['m1', 'm2', 'Mc']
+            # Extract individual variables
+            x, y, z = data[:, 0], data[:, 1], data[:, 2]
+            kde = (data)
+
+            # Plot a 2D contour with a slice along the 'z' dimension
+            kde.plot_2d_contour(parameter[0], parameter[1], slice_dims=[parameter[2]], slice_values=[0], num_points=100)
+
+        """
+        # Generate a grid for the contour plot
+        xx, yy = utils_plot.get_twoD_grid(self.data[dim1], self.data[dim2], num_points=num_points)
+        positions = np.column_stack([xx.ravel(), yy.ravel()])
+
+        #check if slice_dimensions  == dim(KDE) -  2
+        if len(slice_dims) != self.data.shape[1] - 2 :
+            raise ValueError("kde must be computed on more than 2 parameters for slicing")
```

Comment:
ok, but the error message is not very informative.  I would suggest eg `f"With {self.data.shape[1]} KDE dimensions, must specify {self.data.shape[1] - 2} slicing parameters for the plot"`

Also can you put in a check that the length of slice values is equal to the slice parameters

Reply:
replaced